File: hackerrank.com/problem-solving/plus_minus.scala
// https://www.hackerrank.com/challenges/plus-minus/problem

import scala.io.StdIn._

object Solution {

  def plusMinus(arr: Array[Int]) {
    val neg = arr.filter(_ < 0).length.toFloat
    val pos = arr.filter(_ > 0).length.toFloat
    val zer = arr.filter(_ == 0).length.toFloat

    println("%.6f".format(pos / arr.length))
    println("%.6f".format(neg / arr.length))
    println("%.6f".format(zer / arr.length))
  }

  def plusMinus1(arr: Array[Int]) {
    var pos: Double = 0
    var neg: Double = 0
    var zer: Double = 0

    for(i <- arr) {
      if (i < 0)
        neg += 1
      else if (i > 0)
        pos += 1
      else
        zer += 1
    }

    println("%.6f".format(pos / arr.length))
    println("%.6f".format(neg / arr.length))
    println("%.6f".format(zer / arr.length))
  }

  def main(args: Array[String]) {
    val n = readInt
    val values = readLine.split(" ").map(_.trim.toInt)
    plusMinus(values)
    
  }
}

File: hackerrank.com/problem-solving/plus_minus.py
# https://www.hackerrank.com/challenges/plus-minus/problem

from collections import Counter

'''
Elegant solution
Complexity: O(2n)
'''

def plusMinus(arr):
    v = Counter([i // ( abs(i) if i != 0 else 1) for i in arr])
    print ("{:.6f}".format(v[1] / len(arr)))
    print ("{:.6f}".format(v[-1] / len(arr)))
    print ("{:.6f}".format(v[0] / len(arr)))

'''
Brutal force solution
Complexity: O(n)
'''


def plusMinus1(arr):
    pos = neg = zero = 0
    for i in arr:
        if i > 0:
            pos += 1
        elif i < 0:
            neg += 1
        else:
            zero += 1
    print ("{:.6f}".format(pos / len(arr)))
    print ("{:.6f}".format(neg / len(arr)))
    print ("{:.6f}".format(zero / len(arr)))



if __name__ == '__main__':
    n = int(input())
    arr = list(map(int, input().split()))
    plusMinus(arr)
    

  
